Makefile

```diff
@@ -0,0 +1,158 @@
+	@type gofmt    >/dev/null 2>/dev/null || (echo "ERROR: gofmt not found." && false)
+	test -z "$$(gofmt -s -l . | grep -vE '^vendor/|^third_party/' | tee /dev/stderr)"
+	@type golint   >/dev/null 2>/dev/null || (echo "ERROR: golint not found." && false)
+	test -z "$$(golint ./... | grep -v -E '^vendor/|^third_party/' | tee /dev/stderr)"
+	@go doc cmd/vet >/dev/null 2>/dev/null || (echo "ERROR: go vet not found." && false)
+	test -z "$$($(GO) vet $$($(GO) list $(PROJECT)/... | grep -vE '/vendor/|/third_party/') 2>&1 | tee /dev/stderr)"
+
+# Make sure that our builds are reproducible even if you wait between them and
+# the modified time of the files is different.
+local-validate-reproducible:
+	mkdir -p .tmp-validate
+	make -B caasp-init && cp $(BUILD_DIR)/caasp-init .tmp-validate/caasp-init.a
+	@echo sleep 10s
+	@sleep 10s && touch $(GO_SRC)
+	make -B caasp-init && cp $(BUILD_DIR)/caasp-init .tmp-validate/caasp-init.b
+	diff -s .tmp-validate/caasp-init.{a,b}
+	sha256sum .tmp-validate/caasp-init.{a,b}
+	rm -r .tmp-validate/caasp-init.{a,b}
+
+local-validate-build:
+	$(GO) build ${DYN_BUILD_FLAGS} -o /dev/null ${CMD}
+	env CGO_ENABLED=0 $(GO) build ${STATIC_BUILD_FLAGS} -o /dev/null ${CMD}
+	$(GO) test -run nothing ${DYN_BUILD_FLAGS} $(PROJECT)/...
+
+# Used for tests.
+DOCKER_IMAGE :=kubic-project/amd64:tumbleweed
+
+caasp-initimage:
+	docker build -t $(CASSPINIT_IMAGE) .
+
+
+test.unit: caasp-initimage
+	$(DOCKER_RUN) $(CASSPINIT_IMAGE) make test
+
+test: local-validate-go
+	rm -rf /tmp/caasp-init
+	$(GO) test -v ./...
+
+cover:
+	bash <scripts/cover.sh
+
+dist: export COPYFILE_DISABLE=1 #teach OSX tar to not put ._* files in tar archive
+dist:
+	rm -rf build/caasp-init/* release/*
+	mkdir -p build/caasp-init/bin release/
+	cp README.md LICENSE build/caasp-init
+	GOOS=linux GOARCH=amd64 go build -o build/caasp-init/bin/caasp-init -ldflags="$(BASE_LDFLAGS)"
+	tar -C build/ -zcvf $(CURDIR)/release/caasp-init-linux.tgz caasp-init/
+	GOOS=darwin GOARCH=amd64 go build -o build/caasp-init/bin/caasp-init -ldflags="$(BASE_LDFLAGS)"
+	tar -C build/ -zcvf $(CURDIR)/release/caasp-init-macos.tgz caasp-init/
+	rm build/caasp-init/bin/caasp-init
+
+release: dist
+ifndef GITHUB_TOKEN
+	$(error GITHUB_TOKEN is undefined)
+endif
+	github-release release -u kubic-project -r caasp-init --tag $(VERSION)  --name $(VERSION) -s $(GITHUB_TOKEN) -d "$(CHANGE)"
+	github-release upload -u kubic-project -r caasp-init --tag $(VERSION)  --name caasp-init-linux.tgz --file release/caasp-init-linux.tgz -s $(GITHUB_TOKEN)
+	github-release upload -u kubic-project -r caasp-init --tag $(VERSION)  --name caasp-init-macos.tgz --file release/caasp-init-macos.tgz -s $(GITHUB_TOKEN)
+	github-release upload -u kubic-project -r caasp-init --tag $(VERSION)  --name caasp-init-windows.tgz --file release/caasp-init-windows.tgz -s $(GITHUB_TOKEN)
+
+MANPAGES_MD := $(wildcard doc/man/*.md)
+MANPAGES    := $(MANPAGES_MD:%.md=%)
+
+doc/man/%.1: doc/man/%.1.md
+	$(GO_MD2MAN) -in $< -out $@
```

Comment:
I'd recommend taking a look at what we are doing for man pages on [portusctl](https://github.com/openSUSE/portusctl/blob/master/Makefile#L120). That is, md2man usually outputs really awkward man pages (with lots of spaces and so on). As you can see, for that project, I hacked a quick go script that filters unneeded clutter.

Reply:
Nice hack!